src/index.js

```diff
@@ -1,8 +1,463 @@
+const mintTokens = require('./examples/mintTokens');
+const moveFundsBetweenPots = require('./examples/moveFundsBetweenPots');
+const revealTaskWorkRating = require('./examples/revealTaskWorkRating');
+const setTaskBrief = require('./examples/setTaskBrief');
+const setTaskDueDate = require('./examples/setTaskDueDate');
+const setTaskEvaluatorPayout = require('./examples/setTaskEvaluatorPayout');
+const setTaskManagerPayout = require('./examples/setTaskManagerPayout');
+const setTaskRoleUser = require('./examples/setTaskRoleUser');
+const setTaskSkill = require('./examples/setTaskSkill');
+const setTaskWorkerPayout = require('./examples/setTaskWorkerPayout');
+const setTokenOwner = require('./examples/setTokenOwner');
+const signTaskBrief = require('./examples/signTaskBrief');
+const signTaskEvaluatorPayout = require('./examples/signTaskEvaluatorPayout');
+const signTaskWorkerPayout = require('./examples/signTaskWorkerPayout');
+const signTaskDueDate = require('./examples/signTaskDueDate');
+const submitTaskDeliverable = require('./examples/submitTaskDeliverable');
+const submitTaskWorkRating = require('./examples/submitTaskWorkRating');
 
-createColony()
-  .then(createTask)
-  // We're exiting hard here as the providers keep polling otherwise
+// The global database object will act as a mock database where we will store
+// our pending multisig operations so that we can restore the operations when
+// we need to sign them from another account.
+DATABASE = {};
+
+// Test accounts
+const accounts = [
+  '0x90F8bf6A479f320ead074411a4B0e7944Ea8c9C1',
+  '0xFFcf8FDEE72ac11b5c542428B35EEF5769C409f0',
+  '0x22d491Bde2303f2f43325b2108D26f1eAbA1e32b'
+];
+
+// A unix timestamp representing 31 days from now
+const futureDueDate = new Date(Date.now() + 2678400000);
+
+// hackathonStarter example
+const hackathonStarter = async () => {
+
+  // In order to demonstrate the complete task lifecycle, we will need to call
+  // some of the examples using different accounts. The state object below will
+  // store the networkClient and colonyClient for each account as well as state
+  // that will be shared across all accounts.
+  const state = {
+    networkClient: [],              // networkClient (per account)
+    colonyClient: [],               // colonyClient (per account)
+  };
+
+  console.log('\n\x1b[32m' + 'account[0] connectNetwork:' + '\x1b[0m\n');
+
+  // Connect to the network using the "connectNetwork" example and then store
+  // the returned "networkClient" in the state object.
+  state.networkClient[0] = await connectNetwork(
+    0,                              // accountIndex
+  );
+
+  console.log('\n\x1b[32m' + 'account[0] createToken:' + '\x1b[0m\n');
+
+  // Create a new ERC20 token using the "createToken" example and then store
+  // the returned "tokenAddress" in the state object.
+  state.tokenAddress = await createToken(
+    state.networkClient[0],         // networkClient
```

Comment:
A disadvantage of using these functions with arguments like this is that comments need to be added to each argument for clarity here (as opposed to using an object). I'm ok with it, so long as it's maintained as changes are made to the contracts/library.

Reply:
Yup, I agree. I do like that it helps differentiate the examples from the actual colonyJS functions. I think I will leave it as is for now and revisit later. Thanks!